pilot/pkg/model/jwks_resolver.go

```diff
@@ -259,41 +254,56 @@ func (r *jwksResolver) getRemoteContent(uri string) ([]byte, error) {
 		client = r.secureHTTPClient
 	}
 
-	resp, err := client.Get(uri)
-	if err != nil {
-		return nil, err
-	}
-	defer func() {
-		_ = resp.Body.Close()
-	}()
+	getPublicKey := func() ([]byte, error) {
+		resp, err := client.Get(uri)
+		if err != nil {
+			return nil, err
+		}
+		defer func() {
+			_ = resp.Body.Close()
+		}()
+
+		if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+			return nil, fmt.Errorf("unsuccessful response from %q", uri)
+		}
 
-	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
-		return nil, fmt.Errorf("unsuccessful response from %q", uri)
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, err
+		}
+
+		return body, nil
 	}
 
-	body, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return nil, err
+	for i := 0; i < retry; i++ {
+		if body, err := getPublicKey(); err == nil {
+			return body, nil
+		} else {
+			log.Warnf("Failed to fetch JWT public key from %q, retry in %d seconds: %s",
+				uri, jwtPubKeyFetchRetryInSec, err)
```

Comment:
```getRemoteContentWithRetry``` is also used by jwksURI, better update log and var name(jwtPubKeyFetchRetryInSec) to be accurate.

Reply:
updated.